evalml/automl/automl_search.py

```diff
@@ -947,16 +974,15 @@ def full_rankings(self):
 
     @property
     def best_pipeline(self):
-        """Returns an untrained instance of the best pipeline and parameters found during automl search.
+        """If `train_best_pipeline`, returns a trained instance of the best pipeline and parameters found during automl search. Otherwise, returns an untrained pipeline instance.
 
         Returns:
-            PipelineBase: untrained pipeline instance associated with the best automl search result.
+            PipelineBase: If `train_best_pipeline`, trained pipeline instance associated with the best automl search result. Otherwise, untrained pipeline instance.
         """
-        if not self.has_searched:
-            raise PipelineNotFoundError("automl search must be run before selecting `best_pipeline`.")
+        if not (self.has_searched and self._best_pipeline):
+            raise PipelineNotFoundError("automl search must be completely run before selecting `best_pipeline`.")
```

Comment:
Ah, good call.

Nit-pick but I'd remove "completely", not sure if users will understand what that means.